src/runtime_src/core/common/config_reader.cpp

```diff
@@ -171,10 +171,18 @@ get_env_value(const char* env)
 bool
 get_bool_value(const char* key, bool default_value)
```

Comment:
Please revert all changes to config_reader.   Do not hide the exceptions, catch them where you want to deal with them. Coverity points to errors in the main program, not in config_reader.cpp.

Reply:
Thanks @stsoe  for the review. Can you please elaborate on, "coverity points to errors in the main program, not in config_reader.cpp"

Reply:
I see that other APIs like get_string_value()  and get_unit_value() also handles the exceptions in similar way. Hence, I have added catching the exception in this file only.

Reply:
Yes, but that is wrong too.  Only catch exceptions if you are acting on them.  For the Coverity errors you fixed, the problem is all the way in main().  Never write a main() without try/catch

Reply:
Check here.  The error is main.
https://scan9.coverity.com/reports.htm#v37066/p13446/fileInstanceId=62266992&defectInstanceId=7309466&mergedDefectId=208668

Reply:
We need to add below changes in the main() to handle this exception in the root main function. Please let me know if it looks fine?

    try {
      ret = xcldev::xclAwssak(argc, argv);
    } catch (...) {
      std::cout << "Exception caught during xclAwssak()\n";
    }

Reply:
As I mentioned, it would be wrong to try catch these uncaught exceptions in config_reader. Always catch the exceptions where you can take appropriate action;  hiding / ignoring bad user input is not good at all, so it is much better to propagate to main() and take corrective action there.

In general always please catch (const std::exception& ex) and error out with ex.what() so we know what failed and why.  awsak is otherwise almost correct.

However xbmgmt.cpp is not correct.   Lots of code before the later try/catch could throw and is not caught.  Please always write main() c++ programs as something like this:

```
int main(int argc, char* argv[])
{
   try {
     return static_main(argc,argv);  // chose better name 
   }
   catch (const std::exception& ex) {
    xrt_core::send_exception_message(....,..); // or just std::cout
   }
   catch (...) {
     xrt_core::send_exception_message(....); // or just std::cout
  }
  return 1;
}
```

Reply:
Thanks Soren, will make necessary changes for coverity#208670 as you suggested. Remaining coverities fixed in this PR.
I will fix cov#208670 in the next PRs.